<?xml version="1.0" encoding="utf-8"?>	

<launch>

  <arg name="robot_name1" value="ugv1" />
  <arg name="robot_name2" value="ugv2" />
  <arg name="robot_name3" value="ugv3" />
  <arg name="robot_name4" value="ugv4" />
  <arg name="robot_name5" value="ugv5" />
    
  <arg name="simulator" default="/vrep" />
	 <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
      
      
	  <node name="octomap_demux" pkg="octomap_demux" type="octomap_demux5_node" respawn="$(arg respawn_value)" output="screen">
      <param name="global_ref"    value="map"/>
            
      <param name="pcl_sub_name"  value="/volumetric_mapping/octomap_pcl"/>
            
      <param name="pcl1_pub_name" value="$(arg simulator)/$(arg robot_name1)/local_map"/>
			<param name="pcl2_pub_name" value="$(arg simulator)/$(arg robot_name2)/local_map"/>
      <param name="pcl3_pub_name" value="$(arg simulator)/$(arg robot_name3)/local_map"/>
      <param name="pcl4_pub_name" value="$(arg simulator)/$(arg robot_name4)/local_map"/>
      <param name="pcl5_pub_name" value="$(arg simulator)/$(arg robot_name5)/local_map"/>
	  </node> 

</launch>
